### DISCUSSION TITLE
Project fails to start after version upgrade 1.1.12

### DISCUSSION THREAD

**Discussion by blueMountain007** [payloadcms/payload#1260] [Q&A]
# Bug Report

Project fails to start after version upgrade 1.1.12

## Current Behavior

Upgrade the payload to 1.1.12 in order to use the beforeDuplicate hook added in the new version, but the project cannot run

<img width="1239" alt="image" src="https://user-images.githubusercontent.com/16400882/195785435-b0aa3b78-5fae-4a4f-b77b-687fac21147b.png">

<img width="1064" alt="image" src="https://user-images.githubusercontent.com/16400882/195785280-355d25ff-feef-4b8d-9790-0204cc06fcd5.png">

<img width="1627" alt="image" src="https://user-images.githubusercontent.com/16400882/195785340-7fcada62-7436-453a-a3d4-0cf9d2027fba.png">

## Steps to Reproduce

1.
2.
3.
4.

**Comment by jmikrut** [comment]
This seems to be related to a circular dependency in your code somewhere. I can't seem to reproduce. Are you using any custom components? Specifically, are you importing the `Default` Collection Edit view, or the `Relationship` field component?

One thing you can try in your own project is to add the following Webpack plugin:

https://github.com/aackerman/circular-dependency-plugin

And then inject it into your `admin.webpack` config. This will help identify if your code has circular dependencies.  Give that a shot, will you?

I'm gonna convert this to a discussion for now but if we can identify a set of minimally reproducible steps, I will re-open as an issue.

**Reply by blueMountain007** [reply]
@jmikrut  Thank you!

1. the problem has been located; it is the problem of circular dependency. Our collection code references the Relationship component in the payload package (see screenshot 1 below). Our collection is rendered by DefaultEdit (payload/src/admin/components/views/collections/Edit/Default.tsx, hereinafter referred to as DefaultEdit) by default. So far, the reference relationship of the components is as follows: `DefaultEdit -> Relationship`, which is the component Relationship Referenced by the component DefaultEdit.
2. This code commit of #1230; there is logic to reference DefaultEdit in Relationship, and the dependencies here are as follows: `Relationship -> DefaultEdit`.

**The above two dependencies are in opposite directions, forming a circular dependency**

https://github.com/payloadcms/payload/commit/1e048fe03787577fe4d584cec9c2d7c78bc90a17#diff-cea33b3753d693b3965938881d2929f9d2f870882c25354d37509fd6dc6824eaR10

<img width="999" alt="image" src="https://user-images.githubusercontent.com/16400882/196103251-440323ce-c917-458c-8c5c-798e243a14fc.png">
<img width="849" alt="image" src="https://user-images.githubusercontent.com/16400882/196103343-5a82e712-f1bf-49b1-b7d6-09c944eebe44.png">
<img width="534" alt="image" src="https://user-images.githubusercontent.com/16400882/196103446-776875c6-0cb8-4f63-bfab-007526888107.png">
